javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -172,7 +176,20 @@ final class Matcher private[regex] (
     lastMatch
   }
 
-  def groupCount(): Int = ensureLastMatch.length-1
+  def groupCount(): Int = {
+    if (lastMatch != null) lastMatch.length-1
+    else {
```

Comment:
Looking good! One tiny remaining detail: could you format the above as
```scala
if (lastMatch != null) {
  lastMatch.length-1
} else {
  ...
```
to follow [our `if/else` coding style](https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#ifelse) which demands that both branches of an `if/else` must have `{}`, or none of them has braces.

Good to merge after that!